src/components/NewTaxonomy/NewTaxonomy.tsx

```diff
@@ -83,15 +84,40 @@ const NewTaxonomy = ({
     setTreeData(convert(items, options));
   }, [items]);
 
+  const flatten = useMemo(() => {
+    const flatten: TaxonomyItem[] = [];
+    const visitItem = (item: TaxonomyItem) => {
+      flatten.push(item);
+      item.children?.forEach(visitItem);
+    };
+
+    items.forEach(visitItem);
+    return flatten;
+  }, [items]);
+
   const loadData = useCallback(async (node: any) => {
     return onLoadData?.(node.value.split(separator));
   }, []);
 
   return (
     <TreeSelect
       treeData={treeData}
-      value={selected.map(path => path.join(separator))}
-      onChange={items => onChange(null, items.map(item => item.value.split(separator)))}
+      labelInValue={true}
+      value={selected.map(path => {
```

Comment:
better to move this at least to callback function here or maybe even to a method in a model. and then flatten can become a getter in a model, which will increase performance and readability a little.